tests/cmaize/user_api/test_find_or_build_dependency.cmake

```diff
@@ -0,0 +1,79 @@
+# Copyright 2023 CMakePP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include(cmake_test/cmake_test)
+include(cmaize/user_api/cmaize_project)
+include(cmaize/user_api/find_or_build_dependency)
+include(cmaize/utilities/python)
+
+#[[[
+# Tests CMaize's cmaize_find_or_build_dependency function.
+#
+# Note, the innards of the function should be decomposed and tested
+# individually.
+#]]
+ct_add_test(NAME "test_cmaize_find_or_build_dependency")
+function("${test_cmaize_find_or_build_dependency}")
+
+    find_python(py_exe py_version)
+    create_virtual_env(
+        venv_dir "${py_exe}" "${test_cmaize_find_or_build_dependency}"
+    )
+    set(Python3_EXECUTABLE "${venv_dir}/bin/python3")
```

Comment:
For Windows systems, this path is `set(Python3_EXECUTABLE "${venv_dir}/Scripts/python.exe")`.

Reply:
punting on this one.